### PR TITLE
Added support for using a custom scheme in the callback URL [SDK-2223]

### PR DESCRIPTION
**Pull request by Widcket**
### Changes

The Bundle Identifier / Application ID might contain characters not allowed in a URL scheme. This PR adds support for specifying a custom scheme to be used instead.

### Testing

- [X] This change adds unit test coverage
- [ ] This change has been tested on the latest version of the platform/language or why not

### Checklist

- [X] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)
- [X] All existing and new tests complete without errors
- [X] All active GitHub checks have passed

### CODE REVIEW

File: src/webauth/index.js [hunk 1]
```diff
@@ -7,15 +7,17 @@ import verifyToken from '../jwt';
 
 const {A0Auth0} = NativeModules;
 
-const callbackUri = domain => {
+const callbackUri = (domain, customScheme) => {
   const bundleIdentifier = A0Auth0.bundleIdentifier;
   const lowerCasedIdentifier = bundleIdentifier.toLowerCase();
-  if (bundleIdentifier !== lowerCasedIdentifier) {
+  if (!customScheme && bundleIdentifier !== lowerCasedIdentifier) {
```
Comment (lbalmaceda) [inline src/webauth/index.js:13]: I'd not add this condition here. The reason being that when a custom scheme is in use, the case check for the bundle identifier (used not just in the scheme but also in the path) will not be done. And IIRC every callback URL is case sensitive. I think the warning is still helpful to point out possible casing issues. WDYT?

**Review by lbalmaceda — CHANGES_REQUESTED** [review]
Please add some docs on the method and/or in the README to explain how this can be used and in what scenarios it could be useful.

File: src/webauth/index.js (hunk 1 above)
Comment (Widcket) [reply src/webauth/index.js:13]: That's intentional. The bundle identifier is already defined and the developer can't change it, but the custom scheme can be anything the developer wants it to be. So a forced case change is pointless, the developer should be able to specify a valid scheme from the get-go.

**Comment by Widcket** [comment]
> Please add some docs on the method and/or in the README to explain how this can be used and in what scenarios it could be useful.

@lbalmaceda done.

**Comment by lbalmaceda** [comment]
@Widcket Thanks for documenting the parameter. But is not just passing the new option, one would also need to update the intent-filter in the AndroidManifest.xml file, right? 
Developers could benefit from reading that guidance somewhere, e.g. in the readme. I was thinking of adding a small section [like we do here](https://github.com/auth0/react-native-auth0#disable-single-sign-on-ios-13-only) for iOS' SSO. WDYT?

**Review by lbalmaceda — APPROVED** [review]

**Pull request merged**